src/StatsdClient/DogStatsdService.cs

```diff
@@ -16,8 +17,12 @@ public void Configure(StatsdConfig config)
             if (string.IsNullOrEmpty(config.StatsdServerName))
                 throw new ArgumentNullException("config.StatsdServername");
 
+            if (_config != null)
```

Comment:
What do you think of adding a one-line comment to explain why we do this? e.g. race condition in your original PR description.
It's not trivial, and someone with no context may accidentally remove the constraint.

Reply:
> Normally I'm not a big fan of comments describing what the code is already explicitly saying here, I'd prefer to actually add extra documentation to the test case.

I was thinking about a comment that emphasizes the "why" rather than the "what". I think what we are doing is indeed explicit, but the reason why we are doing is not as it requires .
```
// configure once only to avoid race conditions in a multi-threaded context
if (_config != null) 
``` 

> I was pretty sure I added a test case for this, would you accept instead the explanation in the test case where it makes for better executable documentation?

That'd work.